x-pack/dev-tools/simulate_autoscaling.py

```diff
@@ -0,0 +1,276 @@
+from collections import deque, defaultdict
+import enum
+import heapq
+import math
+import random
+import matplotlib.pyplot as plt
+
+
+START_TIME = 0
+END_TIME = 5*3600
+MEASUREMENT_INTERVAL = 1
+
+AUTO_SCALE_UP_THRESHOLD = 0.9
+AUTO_SCALE_DOWN_THRESHOLD = 0.8
+
+AVG_INFERENCE_TIME = 0.1
+PHYSICAL_CORES_PER_NODE = 4
+
+
+random.seed(170681)
+
+
+class Estimator:
```

Comment:
I think it is useful to discuss the basic ideas behind this for anyone else coming to this code
```suggestion
class Estimator:
    """
    This implements a 1 d Kalman filter with manoeuvre detection. Rather than a derived dynamics model 
    we simply fix how much we want to smooth in the steady state.
    """
```